src/net/sourceforge/kolmafia/session/TrainsetManager.java

```diff
@@ -85,7 +85,8 @@ public static TrainsetPiece getByName(String name) {
   private static final Pattern CURRENT_STATION =
       Pattern.compile("<br>Your train is about to pass station (\\d)\\.<");
   private static final Pattern LAPS_BEFORE_RECONFIGURE =
-      Pattern.compile("Let the train finish (\\d) more laps before rearranging it.</p>");
+      Pattern.compile(
```

Comment:
I would have considered doing

```regex
Let the train finish (\d|this)(?: more)? laps? before rearranging it.</p>
```

To make the group matching a little easier to process mentally

Reply:
Or perhaps better
```regex
Let the train finish (\d|this) (?:more laps|lap) before rearranging it.</p>
```